docs/public/components/using-components.md

```diff
@@ -0,0 +1,170 @@
+
+For a component to be instantiated, it must be one of the following:
+
++ Required as a parameter by the `callback` function
++ Required as a dependency from other components already marked for instantiation
++ Directly asked for by using `fx.Invoke`. More on this on the [Fx page](fx.md).
+
+Let's require the `log` components:
+
+```go
+import (
+    // First let's import the FX wrapper to require it
+    logfx "github.com/DataDog/datadog-agent/comp/core/log/fx"
+    // Then the logger interface to use it
+    log "github.com/DataDog/datadog-agent/comp/core/log/def"
+)
+
+// [...]
+    return fxutil.OneShot(
+        myTestCallback, // The function to call from fxutil.OneShot
+        logfx.Module(), // This will tell FX how to create the `log.Component`
+    )
+// [...]
+
+func myTestCallback(logger log.Component) {
+    logger.Info("some message")
+}
+```
+
+## Importing bundles
+
+Now let's say you want to include the core bundle instead. The core bundle offers many basic features (logger, config,
+telemetry, flare, ...).
+
+```go
+import (
+    // We import the core bundle
+    core "github.com/DataDog/datadog-agent/comp/core"
+
+    // Then the interfaces we want to use
+    config "github.com/DataDog/datadog-agent/comp/core/config/def"
+)
+
+// [...]
+    return fxutil.OneShot(
+        myTestCallback, // The function to call from fxutil.OneShot
+        core.Bundle(),  // This will tell FX how to create the all the components included in the bundle
+    )
+// [...]
+
+func myTestCallback(conf config.Component) {
+    api_key := conf.GetString("api_key")
+
+    // [...]
+}
+```
+
+It's very important to understand that since `myTestCallback` only uses the `config.Component`, not all components from
+the `core` bundle are instantiated! The `core.Bundle` instructs Fx how to create components, but only the ones required
+are created.
+
+In our example, the `config.Component` might have dozens of dependencies instantiated from the core bundle. Fx handles
+all of this.
+
+## Using non-component types with Fx
+
+As your migration to components is not finished, you might need to manually instruct Fx on how to create certain types.
+
+You will need to use `fx.Provide` for this. More details can be found [here](fx.md).
+
+But here is a quick example:
+
+```go
+import (
+    logfx "github.com/DataDog/datadog-agent/comp/core/log/fx"
+    log "github.com/DataDog/datadog-agent/comp/core/log/def"
+)
+
+// a non-component type
+type custom struct {}
+
+// [...]
+    return fxutil.OneShot(
+        myTestCallback,
+        logfx.Module(),
+
+        // fx.Provide registers a function providing a type into Fx. Any time this is needed, Fx will use it.
+        fx.Provide(func() custom {
+            return custom{}
+        }),
+    )
+// [...]
+
+// Here our function uses component and non-component type, both provided by Fx.
+func myTestCallback(logger log.Component, c custom) {
+    logger.Info("Custom type: %v", c)
+}
+```
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:**  I have the feeling that `fx.Supply` is more suitable for values which don't require dynamic computing (like Params structs).‏
```suggestion
You will need to use `fx.Supply` for this. More details can be found [here](fx.md).

But here is a quick example:

```go
import (
    logfx "github.com/DataDog/datadog-agent/comp/core/log/fx"
    log "github.com/DataDog/datadog-agent/comp/core/log/def"
)

// a non-component type
type custom struct {}

// [...]
    return fxutil.OneShot(
        myTestCallback,
        logfx.Module(),

        // fx.Supply provides instantiated values into Fx. Any time this is needed, Fx will use it.
        fx.Supply(custom{}),
    )
// [...]

// Here our function uses component and non-component type, both provided by Fx.
func myTestCallback(logger log.Component, c custom) {
    logger.Info("Custom type: %v", c)
}
```
```